src/main/java/bio/terra/flight/dataset/ingest/IngestSetupStep.java

```diff
@@ -49,6 +49,38 @@
     private DatasetService datasetService;
     private BigQueryPdao bigQueryPdao;
 
+    static void validateSourceUri(String sourcePath) {
```

Comment:
Another way to structure this:
- move the wildcard checking and validation into `parseBlobUri` method in IngestUtils.
- add a boolean member in GsUrlParts for "isWildcard" to steer processing later
- deal with file not found inside `bigQueryPdao.loadToStagingTable`

I think that makes it easier for us to use the checking for file ingest later and addresses the TODO below.